plan/physical_plan_builder.go

```diff
@@ -61,6 +62,14 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 		ts.readOnly = true
 	}
 
+	infoSchema, ok := getInfoSchema(p.ctx)
```

Comment:
@tiancaiamao 
We note that if the schema is changed while generating the plan, the schema version number is increased(e.g., 303 --> 304). Since the schema version is one of the components for generating the cache key of the plan, the cache key reflects the schema version of the plan. 
(Please refer to getPstmtKey() in plan/optimizer.go: 86-90)

To reuse the plan, whenever executing the prepared statement, the cache key is generated always. Thus, for example, even though the two prepared statements are same, if their schema versions are different from each other, their cache keys are different from each other. That is, the cache key generated after changing the schema can not hit the plan generated before changing the schema.
(Please refer to OptimizeOnce() in plan/optimizer.go: 58-84)